### PR TITLE
Replace `css` with PostCSS and move test-only package out of dependencies

### PR DESCRIPTION
**Pull request by jecassis**
#### What does this PR do?
Replaced `css` package (last commit Sep 4, 2018) with PostCSS and use it to parse and traverse the AST in `portal.js`. Also leverages Prettier to format the output for the snapshot. Finally, and most importantly, moving this test-only package into `devDependencies` means consumers of Grommet will not see deprecation warnings when installing.

#### Where should the reviewer start?
`src/js/utils/portal.js` for the changes. Then also verify that the differences in the modified snapshot are only whitespace and simply due to Prettier's default CSS formatting.

#### What testing has been done on this PR?
All tests pass. Build completes with almost binary equivalence to 2.13.0.

#### How should this be manually tested?
Snapshots are the only things that should be affected and the diffs should show the change.

#### Any background context you want to provide?
See above.

#### What are the relevant issues?
#4082 

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
Details seem only relevant to developers.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible. No change in behavior or functionality for users.

### CODE REVIEW

**Comment by ShimiSun** [comment]
@jecassis your PRs are awesome! thank you! I'll try to dive into this one soon

File: src/js/utils/portal.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import css from 'css';
+import css from 'postcss'; // eslint-disable-line max-len, import/no-extraneous-dependencies
+import prettier from 'prettier'; // eslint-disable-line max-len, import/no-extraneous-dependencies
```
Comment (ShimiSun) [inline src/js/utils/portal.js:2]: why do we need all the linter comments?

File: src/js/utils/portal.js [hunk 2]
```diff
@@ -10,24 +11,32 @@ export const expectPortal = portalId => ({
...
-          return true;
+        if (rule.selectors) {
+          const selector = rule.selectors.join('');
+          if (
+            !Array.from(node.classList).some(
+              className => selector.indexOf(className) >= 0,
+            )
+          ) {
+            rule.remove();
+          }
```
Comment (ShimiSun) [inline src/js/utils/portal.js:33]: Can you explain what is the motivation for these changes? Is it related to the package changes?

File: src/js/utils/portal.js (hunk 1 above)
Comment (jecassis) [reply src/js/utils/portal.js:2]: The `eslint` rules correctly disallow devDependecies from being imported in the regular source code. However because in `utils` all types of files are thrown in together, some of these are false positives. It is possible to change the global policy or override the rule itself in `.eslintrc` but I thought that was too heavy-handed and could lead to regressions. So I chose to waive the violations per line, however the waiver comments make the line go over the max-len limit so also waived that inline.

Options are:
- Leave as is.
- Waive the block, i.e. disable then enable with the comments above or below.
- Add an additional file to the override block in `.eslintrc` where more rules are also modified or create a new override block just for this file.
- Specify the lint rule globally "specifically" for Grommet throwing away the definition in `eslint-config-airbnb`. If this is what you'd prefer it is better to file it as a new issue. Example:
```json
    'import/no-extraneous-dependencies': ['error', {
      devDependencies: [
        '**/__tests__/**',
        '*.stories.js',
        '**/stories/**'
        '**/portal.js',
        '**/webpack.config.*.js'
      ],
      optionalDependencies: false,
    }],
```

File: src/js/utils/portal.js (hunk 2 above)
Comment (jecassis) [reply src/js/utils/portal.js:33]: Besides the errors, the `css` package does not seem to be maintained any longer and it is not a standalone package so its dependencies are very stale. `postcss` seems to be actively maintained. And so, the motivation, is to migrate Grommet to devDependencies that are maintained with security and bug fixes. `postcss` also greatly expands the capabilities of what can be done given its ecosystem of plugins (though not relevant for this discussion).

This snippet is just reproducing the same behavior of traversing the AST once the CSS is parsed by the tool. Because the specifications for the ASTs are different between `css` and `postcss` the logic needed to be slightly changed. The huge advantage that `postcss` gives for this purpose, in my opinion, is provide OOP-like functions to manipulate the AST instead of interacting directly with its structure.

File: src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap [hunk 3]
```diff
@@ -575,11 +575,12 @@ exports[`Menu close by clicking outside 2`] = `
...
 
 exports[`Menu close by clicking outside 3`] = `
-"@media only screen and (max-width:768px) {
+"@media only screen and (max-width: 768px) {
   .hYQclW {
     padding: 6px;
   }
-}"
+}
+"
```
Comment (ShimiSun) [inline src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap:583]: Can you adjust the logic on portal.js to avoid this line break on the CSS snapshot style? I don't think it is semantically accurate nor necessary and it will fix many of the snapshot changes.

File: src/js/utils/portal.js (hunk 1 above)
Comment (ShimiSun) [reply src/js/utils/portal.js:2]: I'm ok with leaving it as is. thanks for clarifying.

File: src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap (hunk 3 above)
Comment (jecassis) [reply src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap]: Sure. Rebased against `master` and pushed with the update.

File: src/js/components/Drop/__tests__/__snapshots__/Drop-test.js.snap [hunk 4]
```diff
@@ -1383,7 +1364,7 @@ exports[`Drop plain renders 1`] = `
 `;
 
 exports[`Drop plain renders 2`] = `
-"@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {
+"@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
```
Comment (ShimiSun) [inline src/js/components/Drop/__tests__/__snapshots__/Drop-test.js.snap:1367]: I've verified this additional space is the correct behavior, thanks.

File: src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap (hunk 3 above)
Comment (ShimiSun) [reply src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap]: Perfect, thanks buddy!

**Comment by jecassis** [comment]
True, it is much larger. But it only adds cost to the development environment. The output bundle in my local build mirrors almost to the byte and module count that of the latest commit to master. Meaning that `css` is already a test-only dependency. Whether using `css` or `postcss`, moving either out of `dependencies` makes it so that neither `grommet.min.js` nor downstream consumers using es6 modules would see this increase.

**Comment by jecassis** [comment]
Just to add a point to what I said before. PostCSS is a dependency of Storybook and, as such, Grommet was already including it anyway.

**Comment by ShimiSun** [comment]
Thank you for your contribution!
